### ISSUE TITLE
Multiple blueprint registration when blueprint is passed to Api class, this will become an exception in Flask 2.3

### ISSUE THREAD

**Issue by infohash** [python-restx/flask-restx#502]
Run the code below as is.

### **Code**

```python
from flask import Blueprint, Flask
from flask_restx import Api
from flask_restx.resource import Resource

app = Flask(__name__)

blueprint = Blueprint('api', __name__, url_prefix='/api')
api = Api(app=blueprint)
namespace = api.namespace('bug')


@namespace.route('/')
class Bug(Resource):

    @namespace.doc(responses={200: 'OK'})
    def get(self):

        return 'bug'


app.register_blueprint(blueprint)

if __name__ == '__main__':
    app.run()
```

In large scale apps, blueprint is passed to `Api` class and then later all blueprints are lazily registered by Flask app instance by calling `Flask.register()` method. For the purpose of reproducing warning messages, `app`, `blueprint`, `api` and `namespace` instances are together here.

### **Warnings Which Will Later Become An Exception**
```python
UserWarning: The setup method 'add_url_rule' can no longer be called on the blueprint 'api'. It has already been registered at least once, any changes will not be applied consistently.
Make sure all imports, decorators, functions, etc. needed to set up the blueprint are done before registering it.
This warning will become an exception in Flask 2.3.
app_or_blueprint.add_url_rule(self._doc, "doc", self.render_doc)
```

```python
UserWarning: The setup method 'record' can no longer be called on the blueprint 'api'. It has already been registered at least once, any changes will not be applied consistently.
Make sure all imports, decorators, functions, etc. needed to set up the blueprint are done before registering it.
This warning will become an exception in Flask 2.3.
self.record(
```

This will break compatibility with Flask 2.3

### **Expected Behavior**
The warnings have to be fixed before 2.3 is released. Current version of Flask is 2.2.2.

### **Environment**
- Python version - 3.10.9
- Flask version - 2.2.2
- Flask-RESTX version - 1.0.3

**Comment by peter-doggart** [comment]
This looks like a duplicate of #485

**Comment by peter-doggart** [comment]
This should now be fixed in 1.0.5, if not, please open a new issue. :)
